pkg/doc/jwt/jwt.go

```diff
@@ -0,0 +1,448 @@
+		headers: PrepareJWTUnsecuredHeaders(headers),
+		payload: b.claims,
+	}, nil
+}
+
+// New initiates creation of a new JSON Web Tokne. claims could be anything marshallable to the map.
+func New(claims interface{}) Builder {
+	m, err := toMap(claims)
+
+	return &builder{
+		claims: m,
+		err:    err,
+	}
+}
+
+// FromSigned parses JWS in compact serialization format.
+// verifier can be nil, in this case signature verification is not made.
+func FromSigned(jws string, verifier Verifier, detachedPayload []byte) (JSONWebToken, error) {
+	parts := strings.Split(jws, ".")
+	if len(parts) != jwtPartsNumber {
+		return nil, errors.New("invalid JWT compact format")
+	}
+
+	headers, err := base64.RawURLEncoding.DecodeString(parts[jwtHeaderPart])
+	if err != nil {
+		return nil, fmt.Errorf("decode base64 header: %w", err)
+	}
+
+	var jwtHeaders map[string]interface{}
+
+	err = json.Unmarshal(headers, &jwtHeaders)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal JSON headers: %w", err)
+	}
+
+	var payload []byte
+	if len(detachedPayload) == 0 {
+		payload, err = base64.RawURLEncoding.DecodeString(parts[jwtPayloadPart])
+		if err != nil {
+			return nil, fmt.Errorf("decode base64 payload: %w", err)
+		}
+	} else {
+		payload = detachedPayload
+	}
+
+	var jwtPayload map[string]interface{}
+
+	err = json.Unmarshal(payload, &jwtPayload)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal JSON payload: %w", err)
+	}
+
+	signature, err := base64.RawURLEncoding.DecodeString(parts[jwtSignaturePart])
+	if err != nil {
+		return nil, fmt.Errorf("decode base64 signature: %w", err)
+	}
+
+	jwt := &jsonWebToken{
+		headers:   jwtHeaders,
+		payload:   jwtPayload,
+		signature: signature,
+	}
+
+	if verifier != nil {
+		err = verifier.Verify(jwt)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return jwt, nil
+}
+
+// FromUnsecured parses unsecured JWT in compact serialization format.
+func FromUnsecured(jwtUnsecured string, detachedPayload []byte) (JSONWebToken, error) {
+	parts := strings.Split(jwtUnsecured, ".")
+	if len(parts) != jwtPartsNumber {
+		return nil, errors.New("invalid JWT compact format")
+	}
+
+	headers, err := base64.RawURLEncoding.DecodeString(parts[jwtHeaderPart])
+	if err != nil {
+		return nil, fmt.Errorf("decode base64 header: %w", err)
+	}
+
+	var jwtHeaders map[string]interface{}
+
+	err = json.Unmarshal(headers, &jwtHeaders)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal JSON headers: %w", err)
+	}
+
+	var payload []byte
+	if len(detachedPayload) == 0 {
+		payload, err = base64.RawURLEncoding.DecodeString(parts[jwtPayloadPart])
+		if err != nil {
+			return nil, fmt.Errorf("decode base64 payload: %w", err)
+		}
+	} else {
+		payload = detachedPayload
+	}
+
+	var jwtPayload map[string]interface{}
+
+	err = json.Unmarshal(payload, &jwtPayload)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal JSON payload: %w", err)
+	}
+
+	return &jsonWebToken{
+		headers:   jwtHeaders,
+		payload:   jwtPayload,
+		signature: nil,
```

Comment:
FYI: Not needed - zero value.

Reply:
I know :) I just wanted to emphasize this point - that we have no signature in the case of Unsecured JWT.

Reply:
Ok - maybe add a comment too.

Reply:
I removed `nil` assignment.